### PR TITLE
Base scale calculation on icon size, not image size

### LINKED ISSUE #14575: Wrong scale value in cloned icons
**Description**
When an Icons' scale is set as a number and you clone it, the scale of the clone got an array with different values as his scale.
This happens because by calling setScale() an Icon receives an height and width. So by calling the constructor within clone() the scale is set to an array though it was set a number before. This results always in a wrong value for the scale value.
Even the documentation states that height and width can't be used with scale. This seems to be a conflict.

**Steps To Reproduce**
1. Create an Feature with an Icon without height and width
2. Scale the icon with a number Value
3. Clone the icon
4. You will see there is an array as scale value in the cloned style

**Expected behavior**
The cloned icon keeps the same scale as its original especially the same datatype.

### PR DESCRIPTION
**Pull request by ahocevar**
Supersedes #14606 
Supersedes #14601
Fixes #14575

I think this pull request puts things right, given the existing API. See the discussions in the aforementioned issues for more context, but the flesh of this pull request is that the scale calculation from width and height is now based on the icon size, not the image size. Where icon is what gets rendered, and image is the input image, which the rendered icon is a subset from (specified using size and offset).

### CODE REVIEW

**Review by KaiVolland — COMMENTED** [review]
The code looks good to me but the example acts strange.
- Icon does not rerender as expected. 
- Aspect ratio is not as expected.

**Comment by ahocevar** [comment]
@KaiVolland Thanks for the feedback. I'm going to add a rendering test, because - as you can see - an example is not the best way to avoid regressions.

**Review by KaiVolland — APPROVED** [review]
Thanks. For me this is good to go now.

File: src/ol/style/Icon.js [hunk 1]
```diff
@@ -224,27 +244,44 @@ class Icon extends ImageStyle {
...
+            this.setScale(
+              calculateScale(
+                imageSize[0],
+                imageSize[1],
+                options.width,
+                options.height
+              )
+            );
+          };
+          this.listenImageChange(onload);
```
Comment (tschaub) [inline src/ol/style/Icon.js:278]: I expected a return here, excited to find out what happens next.

**Comment by mike-000** [comment]
The example is loading on Chrome and Edge showing an initial scale of 1 while Firefox shows the expected [1.25, 0.83]

**Comment by ahocevar** [comment]
@mike-000:

> The example is loading on Chrome and Edge showing an initial scale of 1 while Firefox shows the expected [1.25, 0.83]

Ignore that. The example uses a non-API way to get the scale, and uses it inappropriately. I'm going to add a rendering test, and remove the example.

**Comment by ahocevar** [comment]
Thanks for the reviews, I have addressed all comments.

**Review by tschaub — APPROVED** [review]

**Pull request merged**